brevets/templates/calc.html

```diff
@@ -0,0 +1,264 @@
+                          checkpoint.find("input[name='close']").val('');
+                          if (km) {
+                              checkpoint_json = {
+                                  "distance": km,
+                                  "location": location,
+                                  "open_time": open,
+                                  "close_time": close
+                              }
+                              checkpoints.push(checkpoint_json);
+                          }
+                      });
+                      var data_to_send = {"start_time": start_time, "brevet_dist": brevet_dist, "checkpoints": checkpoints};
+                      $.ajax({
+                        datatype: 'json',
+                        contentType: "application/json",
+                        type: 'POST',
+                        url: SUBMIT_URL,
+                        data: JSON.stringify(data_to_send),
+                        success: function(data) {
+                            if (data.status == 1) {
+                                console.log("Successfully inserted! Response: ", data);
+                            }
+                            else {
+                                console.log("Insertion failed: ", data);
+                            }
+                         },
+                        error: function(err){
+                         console.log("Error: ", err); }
+                       });
+
+                    });
+
+                $('#fetch').click(
+                   function(event) {
+                       event.preventDefault();
+                       $.getJSON(DISPLAY_URL).done(
+                        function(data) {
+                           console.log(data);
+                            if (data.status == 1) {
+                               console.log("Successfully fetched: ", data);
+                               var start_time = $("#begin_date").val(data.result.start_time);
+                               var brevet_dist = $("#brevet_dist_km").val(data.result.brevet_dist);
+                               var checkpoints = data.result.checkpoints;
+                               var num_checkpoints = checkpoints.length;
+                               $('.control').each(function(index) {
+                                   var lcheckpoint = $(this);
+                                   lcheckpoint.find("input[name='km']").val('');
+                                   lcheckpoint.find("input[name='miles']").val('');
+                                   lcheckpoint.find("input[name='location']").val('');
+                                   lcheckpoint.find("input[name='open']").val('');
+                                   lcheckpoint.find("input[name='close']").val('');
+                                   if (index < num_checkpoints) {
+                                       lcheckpoint.find("input[name='km']").val(checkpoints[index].distance);
```

Comment:
When running your code I noticed that it doesn't displaying miles when loading the data. To do this you can do the following:
```suggestion
                                       lcheckpoint.find("input[name='km']").val(checkpoints[index].distance);
                                       lcheckpoint.find("input[name='miles']").val((0.621371 * checkpoints[index].distance).toFixed(6));
```

This can just calculate the miles on the spot there since we don't save miles.